File: 1798-max-number-of-k-sum-pairs/max-number-of-k-sum-pairs.py
class Solution:
    def maxOperations(self, nums: List[int], k: int) -> int:
        nums.sort()
        i = 0
        j = len(nums) - 1
        max = 0
        
        while i < j:
            if (nums[i] + nums[j] == k):
                max += 1
                i += 1
                j -= 1
            
            elif (nums[i] + nums[j] > k):
                j -= 1

            else:
                i += 1
        return max
                



        

        

File: 1798-max-number-of-k-sum-pairs/max-number-of-k-sum-pairs.cpp
class Solution {
public:
    int maxOperations(vector<int>& nums, int k) {
        sort(nums.begin(), nums.end());
        int max = 0;
        int i = 0;
        int j = nums.size() - 1;
        int result;
        while (i < j) {
            result = nums[i] + nums[j];
            if (result == k) {
                i++;
                j--;
                max++;
            }
            else if (result > k) {
                j--;
            }
            else if (result < k) {
                i++;
            }
        }
        return max;
    }
};